packages/form-section/src/donation-form-section.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  LitElement,
+  html,
+  css,
+  customElement,
+  CSSResult,
+  TemplateResult,
+  property,
+} from 'lit-element';
+
+export enum DonationFormSectionNumberMode {
+  HideNumber = 'hidenumber',
+  ShowNumber = 'shownumber',
+}
+
+@customElement('donation-form-section')
+export class DonationFormSection extends LitElement {
+  @property({ type: Number }) number = 0;
+
+  @property({ type: String }) headline: string | undefined;
+
+  @property({ type: String }) numberMode: DonationFormSectionNumberMode =
+    DonationFormSectionNumberMode.ShowNumber;
+
+  /** @inheritdoc */
+  render(): TemplateResult {
+    return html`
+      <div class="container">
+        <div class="number-container">
+          <div class="number">${this.number}</div>
+        </div>
+        <div class="content-container ${this.contentContainerStyle}">
+          ${this.headline
+            ? html` <div class="title">${this.headline}</div> `
+            : ''}
+          <div class="content">
+            <slot></slot>
+          </div>
+        </div>
+      </div>
+    `;
+  }
+
+  private get contentContainerStyle(): string {
+    switch (this.numberMode) {
+      case DonationFormSectionNumberMode.HideNumber:
+        return 'cover-number';
+      case DonationFormSectionNumberMode.ShowNumber:
+        return '';
+    }
+  }
+
+  /** @inheritdoc */
+  static get styles(): CSSResult {
+    const numberTransition = css`var(--formSectionNumberTransition, 0.25s ease-out)`;
+    const numberMargin = css`var(--formSectionNumberMargin, 1rem)`;
+    const numberBackgroundColor = css`var(--formSectionNumberBackgroundColor, #333)`;
+    const numberRadius = css`var(--formSectionNumberRadius, 1.2rem)`;
+    const numberWidth = css`calc(${numberRadius} * 2)`;
+    const numberFontSize = css`var(--formSectionNumberFontSize, 1.8rem)`;
+    const numberFontWeight = css`var(--formSectionNumberFontWeight, bold)`;
+    const numberFontColor = css`var(--formSectionNumberFontColor, #fff)`;
+    const titleFontSize = css`var(--formSectionTitleFontSize, 1.8rem)`;
+    const titleFontWeight = css`var(--formSectionTitleFontWeight, bold)`;
+
+    const sectionContentBackgroundColor = css`var(--formSectionContentBackgroundColor, white)`;
+
+    const lineHeightCss = css`calc(${numberRadius} * 2)`;
+
+    return css`
+      .container {
+        position: relative;
+        margin-bottom: 1rem;
+      }
+
+      .content-container {
+        position: relative;
+        left: calc(${numberWidth} + ${numberMargin});
+        width: calc(100% - (${numberWidth} + ${numberMargin}));
+        transition: ${numberTransition};
```

Comment:
What facilitated the need for an animated toggle of the order numerals? Or event the toggle in the first place? I can understand optionally rendering the number if it is set, but if this was put in to toggle their display based on available viewport width, some external mechanism in JS would need to constantly listen for resize events and manually adjust the property on each of these components to toggle the number visibility.

Reply:
One of the original designs was going to have the numbers appear and disappear depending on the form state, but we dropped that for now.